activestorage/lib/active_storage/attached/changes/create_one.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  class Attached::Changes::CreateOne #:nodoc:
+    attr_reader :name, :record, :attachable
+
+    def initialize(name, record, attachable)
+      @name, @record, @attachable = name, record, attachable
+    end
+
+    def attachment
+      @attachment ||= find_or_build_attachment
+    end
+
+    def blob
+      @blob ||= find_or_build_blob
+    end
+
+    def upload
+      case attachable
+      when ActionDispatch::Http::UploadedFile, Rack::Test::UploadedFile
```

Comment:
Shouldn't we first check if these constants are defined? Are we sure they're guaranteed to be `require`d by now?

Reply:
> Are we sure they're guaranteed to be required by now?

Yes. They’re required in `lib/active_storage/attached.rb`.

Reply:
Cool, thanks!